### PR TITLE
feat: add Agent and Runtime-Version header interceptors, add release-please

### PR DESCRIPTION
**Pull request by anitarua**
Closes https://github.com/momentohq/dev-eco-issue-tracker/issues/890

Also switches the release workflow to use release-please with the ability to update the package version number in the header interceptor file so the `Agent` header can use the correct version.

### CODE REVIEW

File: internal/grpcmanagers/control_manager.go [hunk 1]
```diff
@@ -24,11 +24,21 @@ func NewScsControlGrpcManager(request *models.ControlGrpcManagerRequest) (*ScsCo
...
 	controlConfig := config.NewStaticGrpcConfiguration(&config.GrpcConfigurationProps{}).WithKeepAliveDisabled()
 
+	headerInterceptors := []grpc.UnaryClientInterceptor{
+		interceptor.AddAuthHeadersInterceptor(authToken),
+	}
+
+	if !interceptor.FirstTimeHeadersSent {
+		interceptor.FirstTimeHeadersSent = true
+		headerInterceptors = append(headerInterceptors, interceptor.AddRuntimeVersionHeaderInterceptor())
+		headerInterceptors = append(headerInterceptors, interceptor.AddAgentHeaderInterceptor("cache"))
```
Comment (pgautier404) [inline internal/grpcmanagers/control_manager.go:34]: Should this be `control` instead?

File: internal/interceptor/first_time_headers_interceptor.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	"context"
+	"fmt"
+	"runtime"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+var FirstTimeHeadersSent = false
+var Version = "1.22.0" // x-release-please-version
```
Comment (bruuuuuuuce) [inline internal/interceptor/first_time_headers_interceptor.go:13]:
```suggestion
var Version = "1.23.1" // x-release-please-version
```

File: .github/workflows/on-push-to-main.yml [hunk 3]
```diff
@@ -79,3 +84,41 @@ jobs:
...
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v3
+
+      - name: Publish package
+        run: |
+          set -e
+          set -x
+          export MOMENTO_SDK_VERSION="${{needs.release-please.outputs.tag_name}}"
+          if [ -z "$MOMENTO_SDK_VERSION"]
```
Comment (bruuuuuuuce) [inline .github/workflows/on-push-to-main.yml:117]:
```suggestion
          if [ -z "$MOMENTO_SDK_VERSION" ]
```

**Review by bruuuuuuuce — APPROVED** [review]
:shipit:

**Pull request merged**